cohttp-async/src/server.mli

```diff
@@ -104,3 +104,31 @@ val create :
   response Async_kernel.Deferred.t) ->
   ('address, 'listening_on) t Async_kernel.Deferred.t
 (** Build a HTTP server, based on the [Tcp.Server] interface *)
+
+module Expert : sig
+  val create :
+    (body:Body.t -> 'addr -> Http.Request.t -> response Async_kernel.Deferred.t) ->
+    'addr ->
+    Async_unix.Reader.t ->
+    Async_unix.Writer.t ->
+    unit Async_kernel.Deferred.t
+  (** [create] accepts a user provided cohttp handler, and creates a server
+      callback that works with user provided socket address,
+      [Async_unix.Reader.t] and [Async_unix.Writer.t]. This can be useful if
+      there is a need for more control over how the Reader and Writer get
+      created. *)
+
+  val create_expert :
```

Comment:
Sorry to nitpick, but this name is a bit tautological. What about `create_with_expert_response`?

Reply:
Good point! I renamed it to `create_with_response_action` as the user can use both a regular response and an expert response here.